deployments/k8s/helm/signalfx-agent/templates/_helpers.tpl

```diff
@@ -55,3 +55,15 @@ Create the name of the secret holding the token
 {{- define "signalfx-agent.secretName" -}}
     {{ default (include "signalfx-agent.fullname" .) .Values.signalFxAccessTokenSecretName }}
 {{- end -}}
+
+{{/*
+Create the configmap name. It will have -v<major version> appended to it from v5 and onward.
+*/}}
+{{- define "configmap-name" -}}
+{{- $major_version := .Values.agentVersion | splitList "." | first | atoi -}}
```

Comment:
Since you don't have to set `agentVersion` (if using custom image), will this blow up in that case?

Reply:
Hmm yeah it's going to fail in that case. Should we make `agentVersion` required then since we're depending on its value to control the update? (ie if they're updating a custom image from v4 to v5, it'd work this time if it goes to v0 but next breaking change it'd still be v0). We also use the `agentVersion` as a pod label. So I think just making it required would be fine, I don't think many people use a custom image anyway.

Reply:
Just use the [default helper](https://helm.sh/docs/chart_template_guide/functions_and_pipelines/#using-the-default-function) and make the default `0.0.0` and then we don't have to worry about all of that.

Reply:
If somebody is using a custom image they don't need us holding their hand on upgrades.